### ISSUE TITLE
Create an individual name note for each field in a JSON form

### ISSUE THREAD

**Issue by joshxg** [openmsupply/mobile#3699]
## The suggestion

- When we fill a survey form for a user, create an individual name note for each field

**Why?**
Dunno

## Describe alternatives you've considered

One name note with a json object with all the data

## Additional context

N/A

**Comment by joshxg** [comment]
@mark-prins 

ugh. sorry.

- I can see that you could index the FK of a name note against the patient event so that there were smaller indicies than indexing the object field (on 4d db, anyway, not 100% sure how pg does object field indicies but it seems you can index individual fields of jsonb?).
- I’m not buying the argument that it makes things more visible - everything is still dynamic and hidden. The fields are just in the type fields of patient events now instead of object fields - the query to get ‘all the fields ever’ is more complex, but no one would be doing that.. maybe querying for the most recent version of the schema, which is just as easy as querying for the types, so I think its a red herring.
- I don’t like our lack of referential integrity - making it better and not worse is the direction I would want to go in ..

Pros of it:
- Easier indicies (kind of, actually if you just index the date and name_ID, you can get all the details of a survey for a patient which has those details)
- Fits the current name note/patient event model better
- Easier to find individual notes (as above, it's easier but alternative is not hard anyway)

Cons:
- Handling object schemas? Just don't add a FK to the schema for any object types? Don't use object schemas except the root?
- Need to also add a 'data type' to name_note and this is indicators with just a row not a row+column!
- Harder to get the full form: becomes a bit of a nightmare (which we need to do when loading a new prescription) - Need to load the schema and traverse it finding all the IDs of patient events, query for all those patient events, and then reconstruct the form object (with nested objects this is pretty shitty?) with the most recent values'

So.. ease of getting the full form v individual fields - I want full form at the application level (for forms, for the list of name notes it's a different story) and individual fields at the dashboard level (Oh, the two databases should be storing the data in two different formats? 🤣 ).

The name_note UI also needs to change, so I think it should just account for this new form being in one of the name_notes.

in conclusion I think we should change it because I can’t be bothered with any “I told you so"'s if anything goes wrong! 😂

**Comment by mark-prins** [comment]
From the points above, I'd say that creating individual notes is not a good idea - repopulating the form alone means that we should retain the full form responses.

> ...the two databases should be storing the data in two different formats?
That's what I keep coming back to - or saving the whole form response _and_ individual name notes for some/all fields.

Trying to think of possible use-cases to determine if we can do what we need to, with only the full form to work with. 
What about:

- Show me all patients who are currently pregnant?
- What is the refusal rate for each village?
- Give me a list of all comorbidities with a count of patients who have reported them

For these we can extract in postgres, so the dashboard is ok. Slightly easier if the data is spread across fields - not worth the pain involved in repopulating the form though.

So what I'm getting to, is that we need the whole form response. and for some things having fields / individual name notes might be helpful. Which means we should stay as we are. And, if later on we find that we have a scenario that needs individual notes, we look at it then.

**Comment by joshxg** [comment]
Going to close this one off
